SwiftMessages/StoryboardSegue.swift

```diff
@@ -0,0 +1,316 @@
+ This class is intended for use with Interface Builder. Reference as `StoryboardSegue` in code.
+ */
+public class CenteredViewSegue: StoryboardSegue {
+    override public  init(identifier: String?, source: UIViewController, destination: UIViewController) {
+        super.init(identifier: identifier, source: source, destination: destination)
+        presentationStyle = .center
+        configure(layout: .centeredView)
+    }
+}
+
+extension StoryboardSegue {
+    public func configure(layout: Layout) {
+        let layer = destination.view.layer
+        switch layout {
+        case .messageView:
+            messageView.bounceAnimationOffset = 5
+            messageView.statusBarOffset = 8
+            messageView.safeAreaTopOffset = -8
+            messageView.safeAreaBottomOffset = -18
+            containment = .content
+            containmentInsets = .zero
+            layer.cornerRadius = 0
+            messageView.configureDropShadow()
+        case .cardView:
+            messageView.bounceAnimationOffset = 0
+            messageView.statusBarOffset = 10
+            messageView.safeAreaTopOffset = -8
+            messageView.safeAreaBottomOffset = -20
+            containment = .background
+            containmentInsets = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
+            layer.cornerRadius = 10
+            layer.masksToBounds = true
+            messageView.configureDropShadow()
+        case .centeredView:
+            messageView.bounceAnimationOffset = 0
+            messageView.statusBarOffset = 10
+            messageView.safeAreaTopOffset = -8
+            messageView.safeAreaBottomOffset = -20
+            containmentInsets = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
+            layer.cornerRadius = 10
+            layer.masksToBounds = true
+            messageView.configureDropShadow()
+        }
+        configuredLayout = layout
+    }
+}
+
+extension StoryboardSegue: UIViewControllerTransitioningDelegate {
+    public func animationController(forPresented presented: UIViewController, presenting: UIViewController, source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        let shower = Shower(segue: self)
+        messenger.defaultConfig.eventListeners.append { [unowned self] in
+            switch $0 {
+            case .didShow:
+                shower.completeTransition?(true)
+            case .didHide:
+                self.hider.completeTransition?(true)
+                self.selfRetainer = nil
+            default: break
+            }
+        }
+        return shower
+    }
+
+    public func animationController(forDismissed dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        return hider
+    }
+}
+
+extension StoryboardSegue {
+    private class Shower: NSObject, UIViewControllerAnimatedTransitioning {
+
+        fileprivate private(set) var completeTransition: ((_: Bool) -> Void)?
```

Comment:
Is the `_: ` in `completeTransition: ((_: Bool) -> Void)?` needed? Can it just be `completeTransition: ((Bool) -> Void)?`?